src/tnc.rs

```diff
@@ -163,7 +168,33 @@ impl Tnc {
             ConnectConfig::TcpKiss(config) => Box::new(TcpKissTnc::open(&config)?),
             ConnectConfig::LinuxIf(config) => Box::new(LinuxIfTnc::open(&config)?),
         };
-        Ok(Self { imp })
+        Ok(Tnc::new(imp))
+    }
+
+    fn new(imp: Box<dyn TncImpl>) -> Self {
+        let tnc = Tnc {
+            imp,
+            senders: Arc::new(Mutex::new(Vec::new())),
+            buffer: Arc::new(Mutex::new(VecDeque::new())),
+        };
+        {
+            let tnc = tnc.clone();
+            // TODO how to allow this thread to exit?
+            thread::spawn(move || {
+                loop {
+                    // TODO what to do if this fails?
+                    if let Ok(x) = tnc.imp.receive_frame() {
```

Comment:
If this fails once, will it always fail? If so, we can just stop looping if there's an error.

Reply:
Yes I think so. I think I made a mistake when I suggested that `incoming()` should return `Receiver<Ax25Frame>` - a better analogy would be `Receiver<Result<Ax25Frame, TncError>>`. This way when an error occurs we can send an `Err(TncError)` and end the loop. That way an application can handle the specific error properly.

In this situation we should also clear our list of `Sender`s (causing the channels to close) in case somebody wrote a downstream application that was simply ignoring any error values. We want their code to realise that no more values are coming so they need to recreate their `Tnc`.